src/modules/services/couchdb.nix

```diff
@@ -0,0 +1,143 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+let
+  cfg = config.services.couchdb;
+  types = lib.types;
+  baseDir = config.env.DEVENV_STATE + "/couchdb";
+  configFile = pkgs.writeText "couchdb.ini" (
+    ''
+      [couchdb]
+      database_dir = ${baseDir}
+      uri_file = ${cfg.uriFile}
+      view_index_dir = ${cfg.viewIndexDir}
+      single_node = true
+
+      [admins]
+      ${cfg.adminUser} = ${cfg.adminPass}
+
+      [chttpd]
+      port = ${toString cfg.port}
+      bind_address = ${cfg.bindAddress}
+      [log]
+      file = ${cfg.logFile}
+    ''
+  );
```

Comment:
What about we adopt this here https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md

Basically we do:

```
settingsFormat = pkgs.formats.ini {};
configFile = settingsFormat.generate "couchdb.ini" cfg.settings;
```

...

```
settings = lib.mkOption {
      type = lib.types.submodule {
        freeformType = settingsFormat.type;

        options.couchdb.database_dir = lib.mkOption {
          type = lib.types.str;
          default = baseDir;
          description = lib.mdDoc ''
            Base Database dir
          '';
        };

        options.couchdb.uri_file = lib.mkOption {
          type = types.path;
          default = "${baseDir}/couchdb.uri";
          description = lib.mdDoc ''
            This file contains the full URI that can be used to access this
            instance of CouchDB. It is used to help discover the port CouchDB is
            running on (if it was set to 0 (e.g. automatically assigned any free
            one). This file should be writable and readable for the user that
            runs the CouchDB service (couchdb by default).
          '';
        };

        options.couchdb.view_index_dir = lib.mkOption {
          type = types.path;
          default = baseDir;
          description = lib.mdDoc ''
            Specifies location of CouchDB view index files. This location should
            be writable and readable for the user that runs the CouchDB service
            (couchdb by default).
          '';
        };

        options.couchdb.single_node = lib.mkOption {
          type = types.bool;
          default = true;
          description = lib.mdDoc ''
            Single node instance
          '';
        };
      };

      default = {};

      description = lib.mdDoc ''
        CouchDB configuration.
      '';
    };
```

then the user is able to configure anything inside the ini file 🤔 . also the options are specified in same position as the real config